ts/index.ts

```diff
@@ -0,0 +1,188 @@
+  FLAKEHUB_PUSH_EXTRA_LABELS?: string;
+  FLAKEHUB_PUSH_SPDX_EXPRESSION?: string;
+  FLAKEHUB_PUSH_ERROR_ON_CONFLICT?: string;
+  FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS?: string;
+  FLAKEHUB_PUSH_ROLLING?: string;
+  FLAKEHUB_PUSH_MIRROR?: string;
+  FLAKEHUB_PUSH_ROLLING_MINOR?: string;
+};
+
+class FlakeHubPushAction {
+  idslib: IdsToolbox;
+
+  // Action inputs translated into environment variables to pass to flakehub-push
+  private tag: string;
+  private host: string;
+  private logDirectives: string;
+  private logger: string;
+  private gitHubToken: string;
+  private repository: string;
+  private directory: string;
+  private gitRoot: string;
+  private extraLabels: string;
+  private spdxExpression: string;
+  private errorOnConflict: boolean;
+  private includeOutputPaths: boolean;
+  private rolling: boolean;
+  private mirror: boolean;
+  private name: string | null;
+  private rollingMinor: number | null;
+
+  constructor() {
+    const options: ActionOptions = {
+      name: "flakehub-push",
+      fetchStyle: "gh-env-style",
+      diagnosticsUrl: new URL(
+        "https://install.determinate.systems/flakehub-push/telemetry",
+      ),
+      legacySourcePrefix: "flakehub-push",
+      requireNix: "fail",
+    };
+
+    this.idslib = new IdsToolbox(options);
+
+    // Inputs translated into environment variables for flakehub-push
+    this.tag = inputs.getString("tag");
+    this.host = inputs.getString("host");
+    this.logDirectives = inputs.getString("log-directives");
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    this.repository = inputs.getString("repository");
+    this.directory = inputs.getString("directory");
+    this.gitRoot = inputs.getString("git-root");
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.rolling = inputs.getBool("rolling");
+    this.mirror = inputs.getBool("mirror");
+    this.name = inputs.getStringOrNull("name");
+    this.rollingMinor = inputs.getNumberOrNull("rolling-minor");
+  }
+
+  // We first check for a value using the `source-binary` input and fall back to the
+  // now-deprecated `flakehub-push-binary`
+  private get sourceBinary(): string | null {
+    const sourceBinaryInput = inputs.getStringOrNull("source-binary");
+    const flakeHubPushBinaryInput = inputs.getStringOrNull(
+      "flakehub-push-binary",
+    );
+
+    return sourceBinaryInput !== ""
+      ? sourceBinaryInput
+      : flakeHubPushBinaryInput;
+  }
+
+  private get visibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
```

Comment:
Do we really need to check this? If we were to ever add another option on the backend, we'd now need to update this instead of just telling people "you can use this new option, `blah`".

I'm fairly certain the backend will validate it anyways.

Reply:
Yeah, probably better to just pass this on to the CLI.

Reply:
Done